aiogram/dispatcher/filters/builtin.py

```diff
@@ -562,3 +562,59 @@ def validate(cls, full_config: typing.Dict[str, typing.Any]) -> typing.Optional[
             return chat_id in self.chat_id
 
         return False
+
+
+class AdminFilter(Filter):
+    """
+    Checks if user is admin in a chat.
+    If is_chat_admin is not set, the filter will check in the current chat (correct only for messages).
+    is_chat_admin is required for InlineQuery.
+    """
+
+    def __init__(self, is_chat_admin: Optional[Union[Iterable[Union[int, str]], str, int, bool]] = None):
+        self._check_current = False
+        self._chat_ids = None
+
+        if is_chat_admin is False:
+            raise ValueError("is_chat_admin cannot be False")
+
+        if is_chat_admin:
+            if isinstance(is_chat_admin, bool):
+                self._check_current = is_chat_admin
+            if isinstance(is_chat_admin, Iterable):
+                self._chat_ids = list(map(int, is_chat_admin))
+            else:
+                self._chat_ids = [int(is_chat_admin)]
+        else:
+            self._check_current = True
+
+    @classmethod
+    def validate(cls, full_config: typing.Dict[str, typing.Any]) -> typing.Optional[typing.Dict[str, typing.Any]]:
+        result = {}
+
+        if "is_chat_admin" in full_config:
+            result["is_chat_admin"] = full_config.pop("is_chat_admin")
+
+        return result
+
+    async def check(self, obj: Union[Message, CallbackQuery, InlineQuery]) -> bool:
+        user_id = obj.from_user.id
+        chat_ids = None
+
+        if self._check_current:
```

Comment:
This block can be optimized to something like that:
```suggestion
        if self._check_current:
            if isinstance(obj, Message):
                message = obj
            elif isinstance(obj, CallbackQuery) and obj.message:
                message = obj.message
            else:
                return False
            if ChatType.is_private(obj.message):  # there is no admin in private chats
                return False
            chat_ids = [obj.message.chat.id]
```
Because you did make duplicated code in lines 606-608 and 610-614

Reply:
Your version is more elegantly than mine